### PR TITLE
Initial commit of the vice analyses view

### PR DESCRIPTION
**Pull request by johnworth**
A database view gathering the information that the VICE feature is going to need/want for it's analysis listing.

### CODE REVIEW

File: src/main/views/08_vice_analyses.sql [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+             c.skip_tmp_mount,
+             p.container_port,
+             i.id AS image_id,
+             i.name AS image_name,
+             i.tag AS image_tag,
+             i.url AS image_url,
+             s.step
+      FROM jobs j
+      JOIN users u ON j.user_id = u.id
+      JOIN job_types t ON j.job_type_id = t.id
```
Comment (psarando) [inline src/main/views/08_vice_analyses.sql]: It doesn't look like any info is used from the `job_types` table, so I think this JOIN can be dropped.

**Review by psarando — COMMENTED** [review]
Usually for listings in the DE, we try to limit the info returned to the minimal required for a usable list, which is usually just the details the user might want to sort the list on, plus sometimes a bit more to display in the listing. Then when the user wants to view more details for a particular item, they can click that item for a detailed view and another endpoint can return all the details for the single item.

The analyses listing can already be pretty slow, so I'm concerned adding all this info this listing could be very slow. I'd recommend copying all the data from all the jobs tables from prod into QA to test the speed of this view.

**Comment by johnworth** [comment]
Alrighty, back to the drawing board.

File: src/main/views/08_vice_analyses.sql (hunk 1 above)
Comment (slr71) [reply src/main/views/08_vice_analyses.sql]: Good catch

**Review by slr71 — COMMENTED** [review]
I tried the listing on a copy of the prod database (on my computer), and the performance wasn't bad. The query to list all of the information in the view took about a quarter second, and most queries against this view are going to be filtering the results further. The performance probably will get a little worse as more interactive jobs are executed, but I don't think it'll be that bad. When we took out the `where` clause and made the join on the `container_port` table a left join, a query to count the total number of jobs in the view (nearly 0.75 million rows) took about 4 seconds. I suspect that the performance will be okay, at least for the time being.

**Comment by psarando** [comment]
OK, that sounds pretty good 👍

**Comment by johnworth** [comment]
Based on the last two comments, I'm going to reopen this pull request.

**Comment by johnworth** [comment]
Since this has been reviewed, doesn't change underlying tables, and is VICE-specific, I'm going to go ahead and merge it. Nothing has changed in the master branch since it was looked over. We should be able to fix it as issues crop up without affecting the DE itself.

**Pull request merged**